### PR TITLE
[processor/groupbytrace] Update metric units

### LINKED ISSUE #34143: Review units of telemetry in metadata.yaml telemetry
### Component(s)

_No response_

### Describe the issue you're reporting

This is the `contrib` equivalent issue for https://github.com/open-telemetry/opentelemetry-collector/issues/10556

Many metrics defined in components' `metadata.yaml` files have the unit of `1`. This isn't meaningful or helpful to users. These metrics should be reviewed to determine a more informative unit.

Here's the list of components that currently have at least one metric with the unit of `1`:

- [ ] `connector/grafanacloudconnector`
- [ ] `connector/servicegraphconnector`
- [ ] `exporter/lokiexporter`
- [ ] `exporter/prometheusremotewriteexporter`
- [ ] `exporter/sumologicexporter`
- [ ] `pkg/stanza/fileconsumer`
- [ ] `processor/filterprocessor`
- [ ] `processor/groupbyattrsprocessor`
- [ ] `processor/groupbytraceprocessor`
- [ ] `processor/k8sattributesprocessor`
- [ ] `processor/probabilisticsamplerprocessor`
- [ ] `receiver/bigipreceiver`
- [ ] `receiver/dockerstatsreceiver`
- [ ] `receiver/elasticsearchreceiver`
- [ ] `receiver/expvarreceiver`
- [ ] `receiver/fluentforwardreceiver`
- [ ] `receiver/hostmetricsreceiver/internal/scraper/cpuscraper`
- [ ] `receiver/hostmetricsreceiver/internal/scraper/filesystemscraper`
- [ ] `receiver/hostmetricsreceiver/internal/scraper/memoryscraper`
- [ ] `receiver/hostmetricsreceiver/internal/scraper/pagingscraper`
- [ ] `receiver/hostmetricsreceiver/internal/scraper/processscraper`
- [ ] `receiver/httpcheckreceiver`
- [ ] `receiver/kafkametricsreceiver`
- [ ] `receiver/kafkareceiver`
- [ ] `receiver/kubeletstatsreceiver`
- [ ] `receiver/mongodbatlasreceiver`
- [ ] `receiver/mongodbreceiver`
- [ ] `receiver/mysqlreceiver`
- [ ] `receiver/oracledbreceiver`
- [ ] `receiver/otelarrowreceiver`
- [ ] `receiver/podmanreceiver`
- [ ] `receiver/postgresqlreceiver`
- [ ] `receiver/redisreceiver`
- [ ] `receiver/snmpreceiver`
- [ ] `receiver/snowflakereceiver`
- [ ] `receiver/solacereceiver`
- [ ] `receiver/sshcheckreceiver`
- [ ] `receiver/statsdreceiver`
- [ ] `receiver/zookeeperreceiver `

### PR DESCRIPTION
**Pull request by honeychaudharyc**
Fix for processor/groupbytraceprocessor component
- [x] ```processor/groupbytraceprocessor```

**Link to tracking Issue:** #34143

### CODE REVIEW

File: processor/groupbytraceprocessor/documentation.md [hunk 1]
```diff
@@ -6,66 +6,66 @@
 
 The following telemetry is emitted by this component.
 
-### otelcol_processor_groupbytrace_conf_num_traces
+### processor_groupbytrace_conf_num_traces
```
Comment (jpkrohling) [inline processor/groupbytraceprocessor/documentation.md:9]: do you know what happened here?

Comment (honeychaudharyc) [reply processor/groupbytraceprocessor/documentation.md:9]: I ran ```go generate ./... ```  and it was updated automatically. I ran this to make sure documentation is up to date with metadata.yaml

Comment (jpkrohling) [reply processor/groupbytraceprocessor/documentation.md]: @codeboten, do you know if this is something concerning?

**Comment by crobert-1** [comment]
I believe this should have a changelog

**Comment by jpkrohling** [comment]
I agree: there should be a changelog for the unit change, and there should not be a change in the metric name. I know @codeboten worked on adding the "otel_" prefix automatically and I think it would reflect in the documentation at the individual components, so this prefix change on this PR looks strange.

**Review by codeboten — COMMENTED** [review]
I think the unit updates look good. As @jpkrohling mentioned, a changelog entry for this would be great. I'm not seeing the prefix change mentioned, has that been addressed?

**Pull request closed without merging**